packages/cluster/src/Envelope.ts

```diff
@@ -0,0 +1,107 @@
+/**
+ * @since 1.0.0
+ */
+import * as RecipientAddress from "@effect/cluster/RecipientAddress"
+import * as Schema from "@effect/schema/Schema"
+import * as Serializable from "@effect/schema/Serializable"
+import * as Data from "effect/Data"
+import type * as Equal from "effect/Equal"
+import * as PrimaryKey from "effect/PrimaryKey"
+
+/**
+ * @since 1.0.0
+ * @category type ids
+ */
+export const TypeId: unique symbol = Symbol.for("@effect/cluster/Envelope")
+
+/**
+ * @since 1.0.0
+ * @category type ids
+ */
+export type TypeId = typeof TypeId
+
+/**
+ * An `Envelope` represents a serializable container for a request that will be
+ * sent to an entity.
+ *
+ * The primary key of an `Envelope` provides the full address of the entity to
+ * which the request should be sent, as well as the identifier of the request,
+ * in the format: `<request-identifier>@<recipient-type>#<entity-identifier>`.
+ *
+ * @since 1.0.0
+ * @category models
+ */
+export interface Envelope<Req extends Schema.TaggedRequest.Any & PrimaryKey.PrimaryKey>
+  extends
+    Equal.Equal,
+    PrimaryKey.PrimaryKey,
+    Serializable.SerializableWithResult<
+      Envelope<Req>,
+      {
+        readonly address: Schema.Schema.Encoded<RecipientAddress.RecipientAddress>
+        readonly message: Schema.Simplify<Schema.Schema.Encoded<Req[typeof Serializable.symbol]>>
+      },
+      Serializable.Serializable.Context<Req>,
+      Schema.Schema.Type<Serializable.WithResult.Success<Req>>,
```

Comment:
`Serializable.WithResult.Success<Req>` returns the actual type, not the schema. Will need to add `.SuccessEncoded` etc.

Reply:
Good call - added those helpers to the `Serializable` module and fixed `Envelope`. Thanks.